modules/converting-machine-api-infra-machines.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc
+
+[id="converting-machine-api-infra-machines_{context}"]
+= Converting infrastructure machines created using the Machine API
+
+[IMPORTANT]
+====
+Creating infra machines using the Machine API is now deprecated.
```

Comment:
What release note are we connecting this deprecation notice to?

Reply:
Good question! I wasn't sure what versions of OCP this decision was targeting. Let me know how/when this deprecation will apply to current versions of OCP (4.5-4.7), and I can add z-stream notices in the release notes where we want this to start.

Reply:
We have to land the deprecation notes (likely) in multiple z' streams all at once. You might be linking this to multiple locations.

Reply:
@sferich888 I can
1. Cherry-pick the updates in this PR on each z-stream release date for the applicable version branch.
2. Draft a release note that I can give the z-stream writers to include in their write-ups for the z-streams we intend to broadcast this for.

Let me know if there's any other tasks to account for.

Reply:
Sorry for the late review, I had a chat with the machine-api team, and they were not 100% onboard with us fully deprecating this method, as with a machineset, you can directly scale an node that has the "infra" label on it.

One compromise we discussed was not fully deprecating the machineset, but instead saying that you must create the infra machineconfigpool to manage your infra nodes. Then you can **also** create an infra machineset such that nodes can be scaled into the pool automatically, albeit with an extra reboot when the node comes up.

Another consideration is, in the future once the MCO has the ability to provision infra nodes directly (the long term plan we talked about), we will likely need some machineset support as well. So in that sense, maybe keeping the machineset (and making notes about the deprecating ONLY having the machineset and not the machineconfigpool) might make more sense.

Again, apologies for the late comment. If you have some questions on the exact phrasing, feel free to ping me on slack and we can work through it

Reply:
I agree with @yuqi-zhang

Reply:
@sferich888 any objections for creating an MCP first and then proceeding with the current strategy of creating infra machine sets? From what I gather, this is the new proposed workflow (anyone correct me if I'm wrong):
1. Create an infra MCP
2. Create infra machine set (as was documented previously)
3. Reboot the provisioned infra nodes, which were scaled into the new infra MCP automatically
4. Strip `worker` label from infra nodes

Reply:
don't forget step 5. set defaultnodeselector on default scheduler to worker label.  (@yuqi-zhang to verify) this would be required to prevent customer workloads from landing on infra nodes.

Reply:
So I did some testing on this. The Infra machineset and MCP does work pretty well together. There is not actually a need for the reboot as described in 3, since its automatic.

If we also want to retain the taint method, the machineset actually support setting a taint on scale, which I just tested to work. So it seems like we have the following options

So basically then it looks like we have the following options to describe a flow:
1. using machinesets + taint
  a) create MCP
  b) create machineset, with an added field that looks like:
```
taints:
      - key: node-role.kubernetes.io/infra
        effect: NoSchedule
```
  Under spec.template.spec.taints of the machineset definition
  c) once that node is up, strip worker label

2. no-machineset method
  a) create infra machinepool
  b) label existing worker as infra. wait for reboot (happens automatically)
  c) remove worker label
  d) taint the node

3. nodeselector method
For either of the above, instead of doing the taint path, we can modify the scheduler instead like Brain suggests.

Which of these do we think we should use moving forward? Are we the right people to discuss this?

Reply:
As a side note, technically 1 and 2 are documented today but both missing some parts of what I described above. We'd probably have to have both, since you cannot use a machineset on e.g. bare metal so for those use cases they'd have to do something manual

Reply:
Oh good to know. Is that also officially covered in OCP docs somewhere? I tried a quick search but it didn't turn up anything

Reply:
@yuqi-zhang that content from the KCS was in-progress for adding to the docs in #28772. However, we were trying to steer away from a lot of that content and were searching for a simpler method to promote (why this current PR was created). This may require additional discussions outside of this PR. I feel we're falling back to where we started a few weeks ago.

Reply:
Ah ok, thanks for the context there. I'd propose then we do the minimal disruptive path in this PR, which is to add the MCP requirement to the machineset method, remove the worker label, but retain the taint (basically, option 1 in above comment). The default scheduler method we can reconsider as an alternative path. Would this be ok?